src/components/app/app.js

```diff
@@ -38,6 +39,9 @@ class App extends React.Component {
     if (prevProps.data !== this.props.data) {
       this.updatePipelineData();
     }
+    if (!isEqual(prevProps.options, this.props.options)) {
```

Comment:
@Huongg `!==` does the shallow comparison and in case of object compare, this checks for reference and not the actual values inside the object. Here in our case `options` is nested object so `isEqual` compares values inside the object and it looks for the changes.